java/src/main/java/io/cucumber/junitxmlformatter/MessagesToJunitXmlWriter.java

```diff
@@ -31,11 +31,19 @@ public MessagesToJunitXmlWriter(OutputStream out) {
     }
 
     public MessagesToJunitXmlWriter(NamingStrategy.ExampleName exampleNameStrategy, OutputStream out) {
-        this(createNamingStrategy(requireNonNull(exampleNameStrategy)), out);
+        this(createNamingStrategy(requireNonNull(exampleNameStrategy), LONG, EXCLUDE), out);
     }
 
-    private static NamingStrategy createNamingStrategy(NamingStrategy.ExampleName exampleName) {
-        return NamingStrategy.strategy(LONG).featureName(EXCLUDE).exampleName(exampleName).build();
+    public MessagesToJunitXmlWriter(NamingStrategy.ExampleName exampleNameStrategy, NamingStrategy.Strategy strategy, OutputStream out) {
+        this(createNamingStrategy(requireNonNull(exampleNameStrategy), requireNonNull(strategy), EXCLUDE), out);
+    }
+
+    public MessagesToJunitXmlWriter(NamingStrategy.ExampleName exampleNameStrategy, NamingStrategy.Strategy strategy, NamingStrategy.FeatureName featureNameStrategy, OutputStream out) {
+        this(createNamingStrategy(requireNonNull(exampleNameStrategy), requireNonNull(strategy), requireNonNull(featureNameStrategy)), out);
+    }
+
+    private static NamingStrategy createNamingStrategy(NamingStrategy.ExampleName exampleName, NamingStrategy.Strategy strategy, NamingStrategy.FeatureName featureName) {
+        return NamingStrategy.strategy(strategy).featureName(featureName).exampleName(exampleName).build();
     }
 
     private MessagesToJunitXmlWriter(NamingStrategy namingStrategy, OutputStream out) {
```

Comment:
@vitalets The alternative would be to make this method public. This way the JS and Java solutions would be similar.

Reply:
That allows the whole naming strategy to be provided. Which isn't great if the test suite and test class names can't also be set because otherwise the `FeatureName#EXCLUDE` should always be used to avoid duplicate names.